Added A sub-menu to change the types of Font and Font-Styles

BarmenduC opened the pull request:
A Menu has been added into the already existing context menu listing various font-families and font-styles that can be used in the note. This caters to #8 and partially to #9 . A few fixes still need to be carried out (would probably be working upon that shortly) .

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -24,6 +24,9 @@
 #undef B_TRANSLATION_CONTEXT
 #define B_TRANSLATION_CONTEXT "View"
 
+
```
Remove stray newlines.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -131,15 +134,22 @@ DeskNoteView::Draw(BRect rct)
-	uint16 currentFace;
-	const rgb_color* bckgrnd;
-	ssize_t size;
+	BAlert 				*alert;
```
Let's keep the style without tabs and added newlines.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -131,15 +134,22 @@ DeskNoteView::Draw(BRect rct)
+					 	ourSize;
+	BMessenger 			*messenger;
+	BMessage			*message;
+	//BFont 			currentFont;
```
Remove commented out code (here and below).

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -186,12 +196,49 @@ DeskNoteView::MessageReceived(BMessage* msg)
 			break;
 		}
+		// Font type
+		case FONT_FAMILY: {
```
Keep to the coding style. Opening bracket in new line.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -186,12 +196,49 @@ DeskNoteView::MessageReceived(BMessage* msg)
+			fontStyle = NULL;
+
+			// Setting the font family
+			msg->FindPointer ("source", &ptr);
```
Keep to the coding style. No space before opening brackets.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -231,6 +278,66 @@ DeskNoteView::MouseDown(BPoint point)
+
+			// Creating a menu of that font family
+			fontMenu = new BMenu (family);
+			fontMenu->SetRadioMode (true);	// I can set only one item as "in use"
```
Don't comment the obvious. Below too.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -320,10 +427,14 @@ DeskNoteView::_ShowContextMenu(BPoint where)
 	colorMenu = new BMenu(B_TRANSLATE("Color"), 0, 0);
 	_BuildColorMenu(colorMenu);
+	
+	styleMenu = new BMenu(B_TRANSLATE("Font Style"));
```
Sentence casing. Though just "Font" would be enough, I suppose.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -370,6 +484,41 @@ DeskNoteView::SaveNote(BMessage* msg)
+		if (superItem != NULL )
+			superItem->SetMarked (true);	// Check the one that was selected
+
+	menuItem = styleMenu->FindItem("Black");
```
Why hardcode "Black" to checkmark?

humdingerb reviewed (changes requested):
* Please use the haiku coding style, as does the rest of the code. There's haiku-format in HaikuDepot that can help with that. I commented a few things, but there are more violations. Also remove additionally inserted newlines.
* The current font menu items aren't checkmarked.
* The replicated view doesn't show the text with the set font.
* Please squash related commits before submitting a PR.

BarmenduC commented:
> * Please use the haiku coding style, as does the rest of the code. There's haiku-format in HaikuDepot that can help with that. I commented a few things, but there are more violations. Also remove additionally inserted newlines.
> * The current font menu items aren't checkmarked.
> * The replicated view doesn't show the text with the set font.
> * Please squash related commits before submitting a PR.

Before performing the code-style cleanup, I'll read up the haiku coding style completely. Would perform the clean up soon enough.

Just to make sure, the tab style for variable declaration is to be dropped everywhere (even in header files) or just inside this file?

The stuff that I mentioned that is still remaining to be fixed are namely these two - 

1. Changed Font not appearing in replicants
2. Checkmark not appearing in-front of selected font

I had planned to keep working upon these issues after this PR gets merged, and to mention these 2 as separate issues so that, in case I am incapable of making it work, someone else can take up the work later. Would that be okay?

humdingerb commented:
> Just to make sure, the tab style for variable declaration is to be dropped everywhere (even in header files) or just inside this file?

Just in the cpp file you changed.

> 1. Changed Font not appearing in replicants
> 2. Checkmark not appearing in-front of selected font
>
> I had planned to keep working upon these issues after this PR gets merged, and to mention these 2 as separate issues so that, in case I am incapable of making it work, someone else can take up the work later. Would that be okay?

I suppose so.
BTW, I recommend getting familiar with using 'git' from the commandline and using feature branches (see https://github.com/haikuports/haikuports/wiki/DevelopmentModel for an explanation). It's much more powerful and even convenient, once you get used to it.

BarmenduC commented:
Requested changes have been mostly fixed. Will be keeping my out for more code-style fixes while I keep working upon  this further.

humdingerb reviewed (changes requested):
Please have another look at the coding style at https://www.haiku-os.org/development/coding-guidelines . As mentioned, "haiku-format" from HaikuDepot can help you.

* Curly brackets at the end of "switch" statements
* Same for "for" and "if" conditions
* Space after "for", "if" etc.
* Space around operators ( =, >= etc)

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -162,13 +156,15 @@ DeskNoteView::MessageReceived(BMessage* msg)
 				"Dropped colors change the background of a note."));
-			const char* extraCopyrights[] = {
+			const char* extraCopyrights[] =
+			{
```
Curly bracket at the end of previous line.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -261,74 +259,74 @@ DeskNoteView::MouseDown(BPoint point)
-					numStyles;
-	BMenu*			fontMenu;
-	uint32			flags;
+	font_family	plainFamily, family;
```
Space insted of tab.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -558,24 +560,26 @@ DeskNoteView::ResizeViewMethod(void* data)
 
-	do {
+	do
+	{
```
Curly bracket at the end of previous line.

BarmenduC commented:
Manual code-style-fixes performed and .haiku-format added

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -156,15 +156,13 @@ DeskNoteView::MessageReceived(BMessage* msg)
 				"Dropped colors change the background of a note."));
-			const char* extraCopyrights[] =
-			{
+			const char* extraCopyrights[] = 
```
Opening curly backet is missing.

humdingerb commented on DeskNoteView.cpp:
```diff
@@ -176,23 +174,25 @@ DeskNoteView::MessageReceived(BMessage* msg)
 			aboutWin->Show();
 			break;
 		}
+		
```
Between every case-statement, there's now a tab. Please remove, no empty line needed between cases.

humdingerb reviewed (changes requested):
Much better. :)
Please remove the haiku.format file. It's only needed somewhere in the folder tree when executing the haiku-format tool.
There are quite a few stray whitespaces around. I recommend installing Koder and "Highlight trailing whitespaces" in its settings. Then they are easily to spot. Maybe even try "Trim trailing whitespaces" from its "Edit" menu.

BarmenduC commented:
Changes performed :-
1. Removed .haiku-format
2. Trimmed trailing whitespaces using Koder
3. Added the missing opening bracket
4. Removed newlines after case statements

BTW, I had a query -> How did the Haiku developers settle on this codestyle (If this is not the right place to ask stuff like this, do tell me where I can do that)(IRC maybe)

humdingerb commented:
> BTW, I had a query -> How did the Haiku developers settle on this codestyle (If this is not the right place to ask stuff like this, do tell me where I can do that)(IRC maybe)

Well, when many people work on a large project for a long time, there needs to be a style guide. The main developers agreed upon the rules very quickly and it changed only very little after that. While the HaikuArchives projects aren't strictly held to these style rules, it is a huge advantage to try have everyone adopt it. Makes it easy to contribute to any project without having to relearn a new style, or god-forbid,everyone doing their own thing and the code becoming a huge smorgasbord of styles...

I'll merge this for now. If it turns out you can't continue working on the open issues, and nobody else does, we can revert the changes.

In the future, please use a feature branch and squash your commits to sensible code bites.
